Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1283 @@
+                    {
+                        col.Add(iface);
+                    }
+                }
+            }
+            else if (NetCollection.TryParse(token, out IPObject obj))
+            {
+                if (!IsIP6Enabled)
+                {
+                    // Remove IP6 addresses from multi-homed IPHosts.
+                    obj.Remove(AddressFamily.InterNetworkV6);
+                    if (!obj.IsIP6())
+                    {
+                        col.Add(obj);
+                    }
+                }
+                else if (!IsIP4Enabled)
+                {
+                    // Remove IP4 addresses from multi-homed IPHosts.
+                    obj.Remove(AddressFamily.InterNetwork);
+                    if (obj.IsIP6())
+                    {
+                        col.Add(obj);
+                    }
+                }
+                else
+                {
+                    col.Add(obj);
+                }
+            }
+            else
+            {
+                _logger.LogDebug("Invalid or unknown network {0}.", token);
+            }
+        }
+
+        /// <summary>
+        /// Handler for network change events.
+        /// </summary>
+        /// <param name="sender">Sender.</param>
+        /// <param name="e">Network availablity information.</param>
+        private void OnNetworkAvailabilityChanged(object? sender, NetworkAvailabilityEventArgs e)
+        {
+            _logger.LogDebug("Network availability changed.");
+            OnNetworkChanged();
+        }
+
+        /// <summary>
+        /// Handler for network change events.
+        /// </summary>
+        /// <param name="sender">Sender.</param>
+        /// <param name="e">Event arguments.</param>
+        private void OnNetworkAddressChanged(object? sender, EventArgs e)
+        {
+            _logger.LogDebug("Network address change detected.");
+            OnNetworkChanged();
+        }
+
+        /// <summary>
+        /// Async task that waits for 2 seconds before re-initialising the settings, as typically these events fire multiple times in succession.
+        /// </summary>
+        /// <returns>The network change async.</returns>
+        private async Task OnNetworkChangeAsync()
+        {
+            try
+            {
+                await Task.Delay(2000).ConfigureAwait(false);
+                InitialiseInterfaces();
+                // Recalculate LAN caches.
+                InitialiseLAN((ServerConfiguration)_configurationManager.CommonConfiguration);
+
+                NetworkChanged?.Invoke(this, EventArgs.Empty);
+            }
+            finally
+            {
+                _eventfire = false;
+            }
+        }
+
+        /// <summary>
+        /// Triggers our event, and re-loads interface information.
+        /// </summary>
+        private void OnNetworkChanged()
+        {
+            if (!_eventfire)
```

Comment:
If the event fires near-simultaneously on separate threads, this will run multiple times.